app/Libraries/Elasticsearch/HasSearch.php

```diff
@@ -20,50 +20,35 @@
 
 namespace App\Libraries\Elasticsearch;
 
-trait HasSearch
+abstract class HasSearch
 {
-    protected $from;
     protected $highlight;
+    protected $params;
     protected $query;
-    protected $size;
-    protected $sorts = [];
     protected $source;
     protected $type;
 
-    /**
-     * @return $this
-     */
-    public function from(?int $from)
+    public function __construct(SearchParams $params)
     {
-        $this->from = $from;
-
-        return $this;
+        $this->params = $params;
     }
 
     /**
      * @return $this
      */
-    public function limit(?int $limit)
+    public function from(int $from)
     {
-        return $this->size($limit);
-    }
-
-    /**
-     * @return $this
-     */
-    public function size(?int $size)
-    {
-        $this->size = $size;
+        $this->params->from = $from;
 
         return $this;
     }
 
     /**
      * @return $this
      */
-    public function page(?int $page)
```

Comment:
`/users/:user/posts` stops working because this is being called in the controller.